invokeai/app/services/invocation_processor/invocation_processor_default.py

```diff
@@ -152,7 +156,8 @@ def __process(self, stop_event: Event):
                     pass
 
                 except CanceledException:
-                    self.__invoker.services.performance_statistics.reset_stats(graph_execution_state.id)
+                    with suppress(GESStatsNotFoundError):
```

Comment:
I made all the stats methods throw a new `GESStatsNotFoundError` exception in the event of an unknown GES ID. This eliminated a lot of null checks when building the structured stats objects.

The only place we use the service is here in the processor. We don't want to stop generation if stats reporting has a problem, so all the places that might throw are wrapped in a `suppress`.

Feels a bit silly, but I'd rather handle the errors here than in the service, where an unknown ID accurately is an error.

Reply:
This makes sense to me.